### PR TITLE
bus-mapping: implement history_hashes and prev. state root

### PR DESCRIPTION
**Pull request by pinkiebell**

### CODE REVIEW

**Comment by pinkiebell** [comment]
> All of these changes are introduced in #825 right?

It was. I somehow messed up my branches but I fixed it. So basically you already reviewed it 😆.

File: bus-mapping/src/circuit_input_builder.rs [hunk 1]
```diff
@@ -287,20 +287,51 @@ impl<P: JsonRpcClient> BuilderClient<P> {
...
+            if prev_state_root.is_none() {
+                prev_state_root = Some(header.state_root.to_word());
+            }
+
+            // latest block hash is the last item
+            let block_hash = header
+                .hash
+                .ok_or(Error::EthTypeError(eth_types::Error::IncompleteBlock))?
+                .to_word();
+            history_hashes[n_blocks] = block_hash;
```
Comment (pinkiebell) [inline bus-mapping/src/circuit_input_builder.rs:323]: @davidnevadoc We may want to review it before merge once we clarify the usage of `history_hashes`. 👍

**Review by han0110 — APPROVED** [review]
LGTM!

**Review by ed255 — APPROVED** [review]
LGTM!

**Pull request merged**